apis/database/v1alpha3/sql_types.go

```diff
@@ -228,89 +228,116 @@ type SQLServerClassList struct {
 // PostgreSQL or MySQL.
 type SQLServerParameters struct {
-
 	// ResourceGroupName specifies the name of the resource group that should
 	// contain this SQLServer.
+	// +immutable
 	ResourceGroupName string `json:"resourceGroupName,omitempty"`
 
 	// ResourceGroupNameRef - A reference to a ResourceGroup object to retrieve
 	// its name
-	ResourceGroupNameRef *ResourceGroupNameReferencerForSQLServer `json:"resourceGroupNameRef,omitempty" resource:"attributereferencer"`
+	// +immutable
+	ResourceGroupNameRef *ResourceGroupNameReferencerForSQLServer `json:"resourceGroupNameRef,omitempty"`
+
+	// SKU is the billing information related properties of the server.
+	SKU SKU `json:"sku"`
 
 	// Location specifies the location of this SQLServer.
+	// +immutable
 	Location string `json:"location"`
 
-	// PricingTier specifies the pricing tier (aka SKU) for this SQLServer.
-	PricingTier PricingTierSpec `json:"pricingTier"`
-
-	// StorageProfile configures the storage profile of this SQLServer.
-	StorageProfile StorageProfileSpec `json:"storageProfile"`
+	// AdministratorLogin - The administrator's login name of a server. Can only be specified when the server is being created (and is required for creation).
+	// +immutable
+	AdministratorLogin string `json:"administratorLogin"`
 
-	// AdminLoginName specifies the administrator login name for this SQLServer.
-	AdminLoginName string `json:"adminLoginName"`
+	// Tags - Application-specific metadata in the form of key-value pairs.
+	// +optional
+	Tags map[string]string `json:"tags,omitempty"`
 
-	// Version specifies the version of this server, for
-	// example "5.6", or "9.6".
+	// Version - Server version.
 	Version string `json:"version"`
 
-	// SSLEnforced specifies whether SSL is required to connect to this
-	// SQLServer.
-	// +optional
-	SSLEnforced bool `json:"sslEnforced,omitempty"`
+	// SSLEnforcement - Enable ssl enforcement or not when connect to server. Possible values include: 'Enabled', 'Disabled'
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	SSLEnforcement string `json:"sslEnforcement"`
+
+	// StorageProfile - Storage profile of a server.
+	StorageProfile StorageProfile `json:"storageProfile"`
 }
 
 // A SQLServerSpec defines the desired state of a SQLServer.
 type SQLServerSpec struct {
 	runtimev1alpha1.ResourceSpec `json:",inline"`
-	SQLServerParameters          `json:",inline"`
+	ForProvider                  SQLServerParameters `json:"forProvider,omitempty"`
 }
 
-// A SQLServerStatus represents the observed state of a SQLServer.
-type SQLServerStatus struct {
-	runtimev1alpha1.ResourceStatus `json:",inline"`
-
-	// State of this SQLServer.
-	State string `json:"state,omitempty"`
-
-	// A Message containing detail on the state of this SQLServer, if any.
-	Message string `json:"message,omitempty"`
+// SQLServerObservation represents the current state of Azure SQL resource.
+type SQLServerObservation struct {
+	// ID - READ-ONLY; Resource ID
```

Comment:
Should we strip these `READ-ONLY` notes out? I think this may be confusing in the context of a Crossplane user reading the CRD or API documentation generated from these comments, given that everything in the observation is read only from their perspective.